tests/integration/lineage.js

```diff
@@ -0,0 +1,489 @@
+    parent: { _id: 'dummyDoc' }
+  }
+};
+const sms_doc = {
+  _id: 'sms_doc',
+  type: 'data_record',
+  from: '123',
+  form: 'D',
+  fields: {},
+  sms_message: {
+    message_id: '76992',
+    message: 'somemessage',
+    from: '123',
+    type: 'sms_message',
+    form: 'D'
+  }
+};
+
+const docs = [
+  circular_area,
+  circular_chw,
+  circular_report,
+  dummyDoc,
+  dummyDoc2,
+  fetch_contact,
+  fetch_noContact,
+  no_contact,
+  no_lineageContact,
+  one_parent,
+  place_parent,
+  place_parentContact,
+  place_grandparent,
+  place_grandparentContact,
+  place_contact,
+  place,
+  report_parent,
+  report_parentContact,
+  report_grandparent,
+  report_grandparentContact,
+  report_contact,
+  report_patient,
+  report,
+  stub_contacts,
+  stub_parents,
+  sms_doc
+];
+const docIds = docs.map(doc => doc._id);
+const deleteDocs = ids => {
+  return db.allDocs({
+    keys: ids,
+    include_docs: true
+  }).then(data => {
+    const docs = data.rows.map(row => {
+      const doc = row.doc;
+      if (doc) {
+        doc._deleted = true;
+        return doc;
+      }
+    }).filter(doc => doc);
+    return db.bulkDocs(docs);
+  }).then(() => {
+    return db.compact();
+  });
+};
+
+describe('Lineage', function() {
+  let lineage;
+
+  before(function() {
+    lineage = lineageFactory(Promise, db);
+    return db.bulkDocs(docs).then(result => {
+      console.log(result);
+      return db.allDocs();
+    }).then(result => {
+      console.log(result);
+    }).catch(error => {
+      console.log('error');
+      console.log(error);
+    });
+  });
+
+  after(function() {
+    return deleteDocs(docIds);
+  });
+
+  afterEach(function() {
+    sinon.restore();
+  });
+
+  it('fetchLineageById queries db with correct parameters', function() {
+    sinon.spy(db, 'query');
+    const id = one_parent._id;
+    return lineage.fetchLineageById(id).then(result => {
+      expect(db.query.callCount).to.equal(1);
+      expect(db.query.getCall(0).args[0]).to.equal('medic-client/docs_by_id_lineage');
+      expect(db.query.getCall(0).args[1].startkey).to.deep.equal([ id ]);
+      expect(db.query.getCall(0).args[1].endkey).to.deep.equal([ id, {} ]);
+      expect(db.query.getCall(0).args[1].include_docs).to.deep.equal(true);
+      expect(result).to.have.lengthOf(2);
+      expect(result[0]).excluding('_rev').to.deep.equal(one_parent);
+      expect(result[1]).excluding('_rev').to.deep.equal(dummyDoc);
+    });
+  });
+
+  it('fetchContacts fetches contacts with correct parameters', function() {
+    sinon.spy(db, 'allDocs');
+    const fakeLineage = [
+      { _id: 'abc', contact: { _id: 'def' }, parent: { _id: 'ghi' } },
+      { _id: 'ghi' }
+    ];
+    return lineage.fetchContacts(fakeLineage).then(() => {
+      expect(db.allDocs.callCount).to.equal(1);
+      expect(db.allDocs.getCall(0).args[0]).to.deep.equal({ keys: ['def'], include_docs: true });
+    });
+  });
+
+  it('fetchHydratedDoc returns errors from query', function() {
```

Comment:
Perhaps more stylish to wrap all the `fetchHydratedDoc()` tests in `describe('#fetchHydratedDoc()', function {`.